### PR TITLE
Fix for issue with video pausing on app pause, resume.

### PR DESCRIPTION
**Pull request by nwestberg**

### CODE REVIEW

**Comment by simonjacobyking** [comment]
Looks good, ship it 👍

File: videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm [hunk 1]
```diff
@@ -7,12 +7,28 @@
 
 @implementation VideoPlayerViewController
 
+static const int INALID_VIDEO_ID = -1;
```
Comment (simonjacobyking) [inline videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm]: Spelling

File: videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm [hunk 2]
```diff
@@ -7,12 +7,28 @@
...
+
+static void QueueVideoCommand(dmVideoPlayer::CommandType commandType, SDarwinVideoInfo& info) {
```
Comment (simonjacobyking) [inline videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm]: Nice 👍

File: videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm [hunk 3]
```diff
@@ -166,57 +213,85 @@ -(void) observeValueForKeyPath:(NSString*)keyPath ofObject:(id)object change:(NS
...
+    if(m_SelectedVideoId != INALID_VIDEO_ID) {
+        SDarwinVideoInfo& info = m_Videos[m_SelectedVideoId];
+        AVPlayer* player = info.m_Player;
+        
+        if(player.status == AVPlayerStatusReadyToPlay && player.currentItem.status == AVPlayerItemStatusReadyToPlay) {
+            [player seekToTime:m_PauseTime toleranceBefore:kCMTimeZero toleranceAfter:kCMTimeZero completionHandler:^(BOOL finished) {
+                [player play];
+            }];
+        } else {
+            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
```
Comment (simonjacobyking) [inline videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm]: "If the player isn't ready, wait a little bit and try resuming again"?

Comment (nwestberg) [reply videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm]: Yes, as per stackoverflow wisdom.

File: videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm (hunk 1 above)
Comment (nwestberg) [reply videoplayer/src/darwin/videoplayer_darwin_viewcontroller.mm]: Oops, thanks. I'll fix that.

**Pull request merged**